scripts/mbedtls_framework/test_case.py

```diff
@@ -89,3 +111,28 @@ def write_data_file(filename: str,
             tc.write(out)
         out.write('\n# End of automatically generated file.\n')
     os.replace(tempfile, filename)
+
+def psa_or_3_6_feature_macro(psa_alg: str,
+                             domain_3_6: str) -> str:
+    """Determine the dependency symbol for a given psa_alg based on
+       the domain and Mbed TLS version. For more information about the domains,
+       and MBEDTLS_MD_CAN_ prefixed symbols, see transition-guards.md.
+    """
+
+    if domain_3_6 == "DOMAIN_3_6_PSA" or not build_tree.is_mbedtls_3_6():
+        if psa_alg in PK_MACROS_3_6 or psa_alg in HASHES_3_6:
+            return psa_alg
+        if psa_alg.startswith('PSA_ALG_') and \
+            psa_alg[8:] in ['MD5', 'RIPEMD160', 'SHA_1', 'SHA_224', 'SHA_256',
+                            'SHA_384', 'SHA_512', 'SHA3_224',
+                            'SHA3_256', 'SHA3_384', 'SHA3_512']:
+            return psa_information.psa_want_symbol(psa_alg)
+
+    if psa_alg in HASHES_3_6:
+        return HASHES_3_6[psa_alg]
+    if psa_information.psa_want_symbol(psa_alg) in HASHES_3_6:
+        return HASHES_3_6[psa_information.psa_want_symbol(psa_alg)]
+    if psa_alg in PK_MACROS_3_6:
+        return PK_MACROS_3_6[psa_alg]
+
+    raise ValueError('Unable to determine dependency symbol for ' + psa_alg)
```

Comment:
Should also print out the domain argument.

(Right now we don't seem to use it: hashes can always use `MD_CAN`, and `PK` is intrinsically in the `USE_PSA` domain.)